packages/amplication-notifications-service/src/core/notifications/notifications.service.ts

```diff
@@ -0,0 +1,38 @@
+import { BadRequestException, Injectable } from '@nestjs/common';
+import { Notification } from 'src/contracts/interfaces/notification.interface';
+import { Novu } from '@novu/node';
+import { ConfigService } from '@nestjs/config';
+import { NotificationMessagePattern } from 'src/contracts/interfaces/notificationMessagePattern.interface';
+
+const NOVU_API_KEY_ENV = 'NOVU_API_KEY';
+
+@Injectable()
+export class NotificationService implements Notification {
+  novuApiKey: string;
+
+  constructor(private configService: ConfigService) {
+    this.novuApiKey = this.configService.get<string>(NOVU_API_KEY_ENV) || '';
+  }
+
+  async pushNotification(
+    notificationData: NotificationMessagePattern
+  ): Promise<void> {
+    const { userId, payload, template } = notificationData;
+
+    const novu = new Novu(this.novuApiKey);
```

Comment:
We can create the novu client in the constructor and store it as a class property. This way we don't need to create a new instance for every notification.